smarts/core/sensors.py

```diff
@@ -1050,3 +1091,33 @@ def closest_point_on_lane(position, lane_id):
 
     def teardown(self):
         pass
+
+
+class TrafficLightSensor(Sensor):
+    def __init__(
+        self, traffic_history: TrafficHistory, map_location_offset: Tuple[float, float]
+    ):
+        self._traffic_history = traffic_history
```

Comment:
fwiw, I think we're eventually going to want to generalize this away from just coming from history data so that SMARTS can manage its own traffic lights, or rely on some other co-simulator (such as Sumo) to do so.  So I'm wondering if it's more appropriate to add some sort of intermediate layer between the sensor and the history data, like a `TrafficLightManager` class or something.  But I guess we can wait on that for now...

Reply:
This for now is just specific to Waymo since Asad wanted this data to be read in. But when TrafficLights will be added and supported by SMARTS in general, we will generalize this as well.